releases/0.6/upgrading/splinter-v0.6-from-v0.4.md

```diff
@@ -0,0 +1,342 @@
+this see [Configuring REST API Authorization]({% link
+docs/0.6/howto/configuring_rest_api_authorization.md %}).
+
+If you were using Biome in 0.4, the only change required is to update the biome
+flag provided to `splinterd`. It has been changed from `--enable-biome` to
+`--enable-biome-credentials`
+
+```
+$ splinterd --enable-biome-credentials
+```
+
+## AdminProtocolVersion has increased
+
+In 0.4, the AdminProtocolVersion was set to 1.
+
+In 0.6, due to several changes in the circuit API the AdminProtocolVersion is
+now set to 2. If you have a client that is compatible with 0.4, make sure to set
+the SplinterProtocolVersion header to 2.
+
+To see the updated API see the [REST API
+reference](https://www.splinter.dev/docs/0.6/api/#tag/Admin-Service)
+
+## Default authorization type is now Challenge Authorization
+
+When Splinter nodes connect, they must go through a “handshake” to verify the
+identity of the other node.
+
+In 0.4, Trust Authorization is the only authorization algorithm implemented.
+Trust Authorization takes the identity provided from the node without further
+verification.
+
+In 0.6, the default authorization type has been updated to Challenge
+Authorization. Challenge Authorization requires a node’s ID to be tied to a
+public key/private key pair and a node must prove they have access to that key
+by signing a random nonce, providing the resulting signature and their public
+key.
+
+This requires that a `splinterd` has its own system key that will be used for
+identification. The key must be stored in the SPLINTER_CONFIG_DIR/keys/. If
+multiple keys exist in this location, all of them will be loaded into
+`splinterd` and can be used for Challenge Authorization. To create this key run:
+
+```
+$ splinter keygen --system
+```
+
+When creating a circuit with the CLI, the public keys for the other members must
+be known at provided. Use `--node-public-key` for each node:
+
+```
+$ splinter circuit propose \
+    --node n20959::tcp://127.0.0.1:18044 \
+    --node n8198::tcp://127.0.0.1:28044 \
+    -node-public-key n20959::0372a7ee5e43a241fb0d622e02a53797507d1b4d289286577157b1ed72a82a6edd \
+    --node-public-key n8198::02bf74d9263327a571763c6557f50d7995bf3dec86387fc8e5f9f75a74b15919a4 \
+    . . .
+```
+
+To continue using Trust authorization, set `--auth-type trust` when proposing
+the circuit.
+
+If using `--peers` on `splinterd` to connect to other nodes without a circuit,
+Challenge Authorization will be used as default also. You can change the key
+that it is used for this authorization with ` --peering-key`. It must be a key
+stored in SPLINTER_CONFIG_DIR/keys/. If you are trying to connect to to a 0.4
+node and need to use trust, append `+trust` to the protocol prefix.
+
+```
+$ splinterd
+     --peering-key other_key \
+    --peers tcp://127.0.0.1:8000 \
+    --peers tcp+trust://127.0.0.1:8001 \
+    . . .
+```
+
+## Certain updates to circuits are not compatible with 0-4
```

Comment:
Change 0-4 to 0.4. "0-4" refers to the git branch, not the version.

Reply:
fixed